ecs-agent/modeltransformer/transformer_test.go

```diff
@@ -0,0 +1,177 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package modeltransformer
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	firstThresholdVersion  = "1.10.0"
+	secondThresholdVersion = "1.20.0"
+)
+
+type Test_task_1_0_0 struct {
+	TestFieldId          string
+	TestFieldContainerId string
+	TestFieldTaskVCpu    string
+}
+
+type Test_task_1_10_0 struct {
+	TestFieldId           string
+	TestFieldContainerIds []string // breaking change introduced in 1.10.0
+	TestFieldTaskVCpu     string
+}
+
+type Test_task_1_20_0 struct {
+	TestFieldId           string
+	TestFieldContainerIds []string
+	TestFieldTaskVCpu     int // breaking change introduced in 1.20.0
+}
+
+func testTransformationFunction1100(dataIn []byte) ([]byte, error) {
+	oldModel := Test_task_1_0_0{}
+	newModel := Test_task_1_10_0{}
+
+	err := json.Unmarshal(dataIn, &oldModel)
+	if err != nil {
+		return nil, err
+	}
+
+	newModel.TestFieldId = oldModel.TestFieldId
+	newModel.TestFieldContainerIds = []string{oldModel.TestFieldContainerId}
+	newModel.TestFieldTaskVCpu = oldModel.TestFieldTaskVCpu
+	dataOut, err := json.Marshal(&newModel)
+	return dataOut, err
+}
+
+func testTransformationFunction1200(dataIn []byte) ([]byte, error) {
+	oldModel := Test_task_1_10_0{}
+	newModel := Test_task_1_20_0{}
+
+	err := json.Unmarshal(dataIn, &oldModel)
+	if err != nil {
+		return nil, err
+	}
+
+	newModel.TestFieldId = oldModel.TestFieldId
+	newModel.TestFieldContainerIds = oldModel.TestFieldContainerIds
+	newModel.TestFieldTaskVCpu, _ = strconv.Atoi(oldModel.TestFieldTaskVCpu)
+	dataOut, err := json.Marshal(&newModel)
+	return dataOut, err
+}
+
+func TestAddTaskTransformationFunctionsAndTransformTask(t *testing.T) {
```

Comment:
Could we add negative test cases? Thanks.

Reply:
What negative test cases are we looking for here? I think the core thing we want to test for transformer is to make sure correct transformation functions are applied, i.e. I think skip one, skip two and not skip is kinda the positive/negative cases

Reply:
Was thinking of test cases like
* If a transformation function has a bug
* If data is corrupted 

And TransformTask() should return an error here (line 68-71 in ecs-agent/modeltransformer/transformer.go).
```
data, err = transformFunc.function(data)
    if err != nil {
        return nil, err
   }
```

Reply:
added